biz.aQute.bndlib/src/aQute/bnd/header/Attrs.java

```diff
@@ -312,12 +316,12 @@ public String put(String key, String value) {
 						}
 						break;
 				}
-				return map.put(attribute, value);
+				return attribute;
 			}
 		}
 		// default String type
 		types.remove(key);
-		return map.put(key, value);
```

Comment:
Why did you change the return type?

Reply:
I split the method in two. One part handles computing the type and modifying the `type` map and the other part modifies the `map` map.